File: 1_TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lengthList = []
        for i in nums:
            lengthList.append(target - i)
        print(lengthList)
        for i in range(len(lengthList)):
            if lengthList[i] in nums and (nums.index(lengthList[i]) != i):
                return [i, nums.index(lengthList[i])]



s = Solution()
print(s.twoSum([3, 3, 4], 6))

File: 1_TwoSum.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        
        for (i, &num) in nums.iter().enumerate() {
            let complement = target - num;
            if let Some(&prev_index) = map.get(&complement) {
                return vec![prev_index as i32, i as i32];
            }
            map.insert(num, i);
        }
        
        vec![-1, -1] // Return [-1, -1] if no solution found
    }
}